src/monitor/poll_monitor.cpp

```diff
@@ -19,58 +19,80 @@
  * along with lo2s.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include <lo2s/monitor/fd_monitor.hpp>
+#include <lo2s/config.hpp>
+#include <lo2s/error.hpp>
+#include <lo2s/monitor/poll_monitor.hpp>
 
-#include <nitro/lang/enumerate.hpp>
+extern "C"
+{
+#include <sys/timerfd.h>
+}
 
 namespace lo2s
 {
 namespace monitor
 {
-FdMonitor::FdMonitor(trace::Trace& trace, const std::string& name) : ThreadedMonitor(trace, name)
+PollMonitor::PollMonitor(trace::Trace& trace, const std::string& name)
+: ThreadedMonitor(trace, name)
 {
-    pfds_.resize(1);
+    pfds_.resize(2);
     stop_pfd().fd = stop_pipe_.read_fd();
     stop_pfd().events = POLLIN;
     stop_pfd().revents = 0;
+
+    // Create and initialize timer_fd
+    struct itimerspec tspec;
+    memset(&tspec, 0, sizeof(struct itimerspec));
+
+    // Set initial expiration to lowest possible value, this together with TFD_TIMER_ABSTIME should
+    // synchronize our timers
+    tspec.it_value.tv_nsec = 1;
+    tspec.it_interval.tv_nsec = config().read_interval.count();
+
+    timer_pfd().fd = timerfd_create(CLOCK_MONOTONIC, 0);
+    timer_pfd().events = POLLIN;
```

Comment:
make the timerfd `O_NONBLOCK` to ensure the `read` during flush does *never* block.